### PR TITLE
fix: make add_piece taking less time

### PR DESCRIPTION
**Pull request by vmx**
`add_piece` operates in 64 bytes pieces. It's faster to operate in bigger chunks. This commits increases the buffer size to 4KiB. This makes adding a 32GiB piece about more than 2x faster. On the hardware I used it goes down from 14min to about 6min.

### CODE REVIEW

File: filecoin-proofs/src/commitment_reader.rs [hunk 1]
```diff
@@ -7,11 +7,13 @@ use rayon::prelude::{ParallelIterator, ParallelSlice};
 
 use crate::{constants::DefaultPieceHasher, pieces::piece_hash};
 
+const BUFFER_SIZE: usize = 4096;
```
Comment (Kubuxu) [inline filecoin-proofs/src/commitment_reader.rs:10]: Have you tried bumping it even further, for example, up to 1MiB-8MiB?
Even with the 4KiB, it is just 64 SHA invocations, the branch predictor is probably just warming up at that point.

Comment (vmx) [reply filecoin-proofs/src/commitment_reader.rs:10]: I've tried 8MiB it didn't make a difference.

I think yet another difference would make it if the hashing would be done in parallel. But a simple "par chunker" didn't really help, as then the pieces are to small. One need to implement it manually. I decided that it's not worth it and that this improvement is already good enough for how simple it is.

**Comment by cryptonemo** [comment]
Can we check if lotus is using this method?  I recall it not being used, but may be wrong there.  In any case, it should help speed up our big tests.

**Comment by cryptonemo** [comment]
> > Can we check if lotus is using this method?
> 
> `write_with_alignment` seems to be the only API call in the FFI using it. It is then called `WriteWithAlignment` on the Go side. If I grep the Lotus repo for that, I only find it used in tests.
> 
> Why is that important? It's not a breaking change, it just improves things. So if Lotus would use it, it would also be beneficial for them.

I just wanted to know if we'd see the speed-ups outside of tests.

**Review by cryptonemo — APPROVED** [review]

**Pull request merged**